src/neuroconv/datainterfaces/ophys/inscopix/inscopiximagingdatainterface.py

```diff
@@ -0,0 +1,197 @@
+        if device_info:
+            device_metadata = metadata["Ophys"]["Device"][0]
+
+            # Update the actual device name
+            if device_info.get("device_name"):
+                device_metadata["name"] = device_info["device_name"]
+
+            # Build device description
+            microscope_info = []
+            if device_info.get("device_serial_number"):
+                microscope_info.append(f"Serial: {device_info['device_serial_number']}")
+            if device_info.get("acquisition_software_version"):
+                microscope_info.append(f"Software: {device_info['acquisition_software_version']}")
+
+            if microscope_info:
+                device_metadata["description"] = f"Inscopix Microscope ({', '.join(microscope_info)})"
+
+            # Update imaging plane metadata with acquisition details
+            imaging_plane_metadata = metadata["Ophys"]["ImagingPlane"][0]
+
+            # Update imaging plane device reference to match the actual device name
+            if device_info.get("device_name"):
+                imaging_plane_metadata["device"] = device_info["device_name"]
+
+            acquisition_details = []
+            if device_info.get("exposure_time_ms"):
+                acquisition_details.append(f"Exposure Time (ms): {device_info['exposure_time_ms']}")
+            if device_info.get("microscope_gain"):
+                acquisition_details.append(f"Gain: {device_info['microscope_gain']}")
+            if device_info.get("microscope_focus"):
+                acquisition_details.append(f"Focus: {device_info['microscope_focus']}")
+            if device_info.get("efocus"):
+                acquisition_details.append(f"eFocus: {device_info['efocus']}")
+            if device_info.get("led_power_ex_mw_per_mm2"):
+                acquisition_details.append(f"EX LED Power (mw/mm^2): {device_info['led_power_ex_mw_per_mm2']}")
+            if device_info.get("led_power_og_mw_per_mm2"):
+                acquisition_details.append(f"OG LED Power (mw/mm^2): {device_info['led_power_og_mw_per_mm2']}")
+
+            if acquisition_details:
+                current_description = imaging_plane_metadata.get(
+                    "description", "The plane or volume being imaged by the microscope."
+                )
+                imaging_plane_metadata["description"] = f"{current_description} ({'; '.join(acquisition_details)})"
+
+        # Subject information
+        subject_metadata = {}
+        has_any_subject_data = False
+
+        # Subject ID
+        if subject_info and subject_info.get("animal_id"):
+            subject_metadata["subject_id"] = subject_info["animal_id"]
+            has_any_subject_data = True
+
+        species_value = None
+        strain_value = None
+
+        if subject_info and subject_info.get("species"):
+            species_raw = subject_info["species"]
+            # If it contains genotype info or matches NWB format, put it in species; otherwise strain
```

Comment:
Can you write examples of these field looks in the metadata for the examples that we have so far. It will be easier for me to parse the logic of this if.

Reply:
Ok.